info/diktat-kotlin-coding-style-guide-en.md

```diff
@@ -1641,3 +1642,249 @@ Good example:
      // ...
  }
 ``` 
+
+
+# <a name="c6"></a>6 Classes, interfaces and functions
+### <a name="c6.1"></a>6.1 Classes
+### <a name="r6.1.1"></a> Rule 6.1.1: Primary constructor should be define implicitly in the declaration of class
+In case class contains only one explicit constructor - it should be converted to implicit primary constructor.
+Bad example:
+```kotlin
+class Test {
+    var a: Int
+    constructor(a: Int) {
+        this.a = a
+    }
+}
+```
+
+Good example:
+```kotlin
+class Test(var a: Int) { 
+    // ...
+}
+
+// in case of any annotations or modifer used on constructor:
+class Test private constructor(var a: Int) { 
+    // ...
+}
+```
+
+### <a name="r6.1.2"></a> Rule 6.1.2: Prefer data classes instead of classes without any functional logic
+Some people say that data class - is a code smell, but in case you really need to use it and your is becoming more simple because of that -
```

Comment:
```suggestion
Some people say that data class - is a code smell, but in case you really need to use it and your code is becoming more simple because of that -
```
I thought data classes are one of the killer-features of kotlin, do we really need to mention in our code style that it is considered a code smell?

Reply:
Yeah, that's why I say "some people" say :))
https://refactoring.guru/smells/data-class